tags/serializers.py

```diff
@@ -0,0 +1,70 @@
+from rest_framework import serializers
+from rest_framework.relations import PrimaryKeyRelatedField
+
+from tags.models import Tag, Namespace
+
+
+class NamespaceSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Namespace
+        fields = ("id", "name", "scope", "scoped_to")
+
+    def validate_scope(self, value):
+        if value not in Namespace.SCOPES.keys():
+            raise serializers.ValidationError("The scope isn't valid")
+        return value
+
+    def validate(self, data):
+        if len(data) == 0:
+            return data
+
+        if bool(Namespace.SCOPES.get(data.get("scope", None), False)) != bool(
+            data.get("scoped_to", False)
+        ):  # logical XOR
```

Comment:
Weird formatting here (also in the next tests)

Reply:
Black formats as it, let's trust it